src/utils/daos_hdlr.c

```diff
@@ -2948,6 +2963,467 @@ fs_copy_hdlr(struct cmd_args_s *ap)
+			 * type
+			 */
+			if ((int)iod.iod_size == 0) {
+				rc = cont_copy_recx_array(&diov, &aiov, src_oh,
+							  dst_oh, &iod);
+				if (rc != 0) {
+					fprintf(stderr, "failed to copy "
+						"record: %d\n", rc);
+					D_GOTO(out, rc);
+				}
+			} else {
+				rc = cont_copy_recx_single(&diov, src_oh,
+							   dst_oh, &iod);
+				if (rc != 0) {
+					fprintf(stderr, "failed to copy "
+						"record: %d\n", rc);
+					D_GOTO(out, rc);
+				}
+			}
+			/* advance to next akey returned */
+			ptr += akey_kds[j].kd_key_len;
+		}
+	}
+out:
+	return rc;
+}
+
+static int
+cont_copy_list_dkeys(daos_handle_t *src_oh,
+		     daos_handle_t *dst_oh)
+{
+	int		rc = 0;
+	int		j = 0;
+	int		buf_len = 0;
+	char		*ptr;
+	daos_anchor_t	dkey_anchor = {0};
+	daos_key_desc_t	dkey_kds[ENUM_DESC_NR] = {0};
+	uint32_t	dkey_number = ENUM_DESC_NR;
+	char		dkey_enum_buf[ENUM_DESC_BUF] = {0};
+	char		dkey[ENUM_KEY_BUF] = {0};
+	d_sg_list_t	sgl;
+	d_iov_t		iov;
+	daos_key_t	diov;
+
+	/* loop to enumerate dkeys */
+	while (!daos_anchor_is_eof(&dkey_anchor)) {
+		memset(dkey_kds, 0, sizeof(dkey_kds));
+		memset(dkey_enum_buf, 0, sizeof(dkey_enum_buf));
+		memset(dkey, 0, sizeof(dkey));
+		dkey_number = ENUM_DESC_NR;
+
+		sgl.sg_nr     = 1;
+		sgl.sg_nr_out = 0;
+		sgl.sg_iovs   = &iov;
+
+		d_iov_set(&iov, dkey_enum_buf, ENUM_DESC_BUF);
+
+		/* get dkeys */
+		rc = daos_obj_list_dkey(*src_oh, DAOS_TX_NONE,
+					&dkey_number, dkey_kds,
+					&sgl, &dkey_anchor, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to list dkeys: %d\n", rc);
+			D_GOTO(out, rc);
+		}
+
+		/* if no dkeys were returned move on */
+		if (dkey_number == 0)
+			continue;
+
+		/* parse out individual dkeys based on key length and
+		 * number of dkeys returned
+		 */
+		for (ptr = dkey_enum_buf, j = 0; j < dkey_number; j++) {
+			/* Print enumerated dkeys */
+			buf_len = snprintf(dkey, dkey_kds[j].kd_key_len + 1,
+					   "%s", ptr);
+			if (buf_len >= ENUM_KEY_BUF) {
+				fprintf(stderr, "dkey is too large: %d\n", rc);
+				D_GOTO(out, rc = 1);
+			}
+			d_iov_set(&diov, (void *)dkey, dkey_kds[j].kd_key_len);
+
+			/* enumerate and parse akeys */
+			rc = cont_copy_list_akeys(src_oh, dst_oh, diov);
+			ptr += dkey_kds[j].kd_key_len;
+		}
+	}
+out:
+	return rc;
+}
+
+static int
+cont_copy_parse(struct duns_attr_t *dattr,
+		uuid_t *pool,
+		uuid_t *cont,
+		char *arg)
+{
+	int	rc = 0;
+	char	*saveptr = NULL;
+	char	*pool_str = NULL;
+	char	*cont_str = NULL;
+
+	/* check for DAOS pool/cont or UNS path */
+	rc = duns_resolve_path(arg, dattr);
+	if (rc != 0) {
+		pool_str = strtok_r(arg, "/", &saveptr);
+		rc = uuid_parse(pool_str, *pool);
+		if (rc == 0) {
+			cont_str = strtok_r(NULL, "/", &saveptr);
+			rc = uuid_parse(cont_str, *cont);
+			if (rc != 0) {
+				fprintf(stderr, "failed to parse cont "
+					"uuid: %d\n", rc);
+				D_GOTO(out, rc);
+			}
+		} else {
+			fprintf(stderr, "failed to parse uuid or "
+				"path: %d\n", rc);
+			D_GOTO(out, rc = EINVAL);
+		}
+	} else {
```

Comment:
@dsikich Future improvement:
`duns_resolve_path` can also parse the pool/cont uuids as well. And you can set
`attr->da_no_prefix` to tell it whether or not the path has `daos:` prefix